src/metabase/query_processor/streaming/csv.clj

```diff
@@ -35,6 +36,38 @@
   Disabled by default and should remain disabled until Issue #44556 is resolved and a clear plan is made."
   false)
 
+(defn- write-csv
+  "Custom implementation of `clojure.data.csv/write-csv` with a more efficient quote? predicate and no support for
+  options (we don't use them)."
+  [writer data]
+  (let [separator \,
+        quote \"
+        quote? (fn [^String s]
+                 (let [n (.length s)]
+                   (loop [i 0]
+                     (if (>= i n) false
+                         (let [ch (.charAt s (unchecked-int i))]
+                           (if (or (= ch \,) ;; separator
+                                   (= ch \") ;; quote
```

Comment:
Why are you referring to the literals here, as opposed to the bindings? Just seems more fragile if we tweak them.

Reply:
Java would implicitly make them final primitives, so I hoped as much for Clojure 🥹 

Good call to be cautious.

Reply:
I've checked; the closed over Characters are correctly downcasted to chars when comparing, so it's all OK.